src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-form/osd-form.component.ts

```diff
@@ -209,16 +246,50 @@ export class OsdFormComponent extends CdForm implements OnInit {
     }
   }
 
+  emitDeploymentSelection() {
+    const options = document.querySelector(
+      'input[name="deploymentOptions"]:checked'
+    ) as HTMLInputElement;
+    this.emitDeploymentOption.emit(options?.value);
+  }
+
   submit() {
-    // use user name and timestamp for drive group name
-    const user = this.authStorageService.getUsername();
-    this.driveGroup.setName(`dashboard-${user}-${_.now()}`);
-    const modalRef = this.modalService.show(OsdCreationPreviewModalComponent, {
-      driveGroups: [this.driveGroup.spec]
-    });
-    modalRef.componentInstance.submitAction.subscribe(() => {
-      this.router.navigate(['/osd']);
-    });
-    this.previewButtonPanel.submitButton.loading = false;
+    if (this.simpleDeployment) {
+      const options = document.querySelector(
+        'input[name="deploymentOptions"]:checked'
+      ) as HTMLInputElement;
+      const deploymentSpec = { option: options.value };
+      const trackingId = `${deploymentSpec.option} optmized deployment`;
+      this.taskWrapper
+        .wrapTaskAroundCall({
+          task: new FinishedTask('osd/' + URLVerbs.CREATE, {
+            tracking_id: trackingId
+          }),
+          call: this.osdService.create([deploymentSpec], trackingId, 'predefined')
+        })
+        .subscribe({
+          complete: () => {
+            this.router.navigate(['/osd']);
+          }
+        });
+    } else {
+      // use user name and timestamp for drive group name
+      const user = this.authStorageService.getUsername();
+      this.driveGroup.setName(`dashboard-${user}-${_.now()}`);
+      const modalRef = this.modalService.show(OsdCreationPreviewModalComponent, {
+        driveGroups: [this.driveGroup.spec]
+      });
+      modalRef.componentInstance.submitAction.subscribe(() => {
+        this.router.navigate(['/osd']);
+      });
+      this.previewButtonPanel.submitButton.loading = false;
+    }
+  }
+
+  submitText() {
+    if (this.simpleDeployment) {
+      return $localize`Create OSDs`;
+    }
+    return this.actionLabels.PREVIEW;
```

Comment:
Everytime I see translations in a typescript file I feel that something could be done better. Can't we have this inside the HTML template? This looks like an `*ngIf` inside a button?

BTW, just out of curiosity, add a `console.error("I'm here")` to this and please tell me how many times you see this message in the console log?

Reply:
This was also an accident. I used it for testing purposes and forgot to update it.